guide/src/types.md

```diff
@@ -1,53 +1,47 @@
-# GIL lifetimes, mutability and Python object types
+# Python object types
 
-On first glance, PyO3 provides a huge number of different types that can be used
-to wrap or refer to Python objects.  This page delves into the details and gives
-an overview of their intended meaning, with examples when each type is best
-used.
+PyO3 offers two main sets of types to interact with Python objects. This section of the guide expands into detail about these types and how to choose which to use.
+
+The first set of types is are the "smart pointers" which all Python objects are wrapped in. These are `Py<T>`, `Bound<'py, T>`, and `Borrowed<'a, 'py, T>`. The [first section below](#pyo3s-smart-pointers) expands on each of these in detail and why there are three of them.
+
+The second set of types are types which fill in the generic parameter `T` of the smart pointers. The most common is `PyAny`, which represents any Python object (similar to Python's `typing.Any`). There are also concrete types for many Python built-in types, such as `PyList`, `PyDict`, and `PyTuple`. User defined `#[pyclass]` types also fit this category. The [second section below](#concrete-python-types) expands on how to use these types.
+
+Before PyO3 0.21, PyO3's main API to interact with Python objects was a deprecated API known as the "GIL Refs" API, containing reference types such as `&PyAny`, `&PyList`, and `&PyCell<T>` for user-defined `#[pyclass]` types. The [third section below](#the-gil-refs-api) details this deprecated API.
+
+## PyO3's smart pointers
 
+PyO3's API offers three generic smart pointers: `Py<T>`, `Bound<'py, T>` and `Borrowed<'a, 'py, T>`. For each of these the type parameter `T` will be filled by a [concrete Python type](#concrete-python-types). For example, a Python list object can be represented by `Py<PyList>`, `Bound<'py, PyList>`, and `Borrowed<'a, 'py, PyList>`.
 
-## The Python GIL, mutability, and Rust types
+These smart pointers have different numbers of lifetime parameters, which defines how they behave. `Py<T>` has no lifetime parameters, `Bound<'py, T>` has a lifetime parameter `'py`, and `Borrowed<'a, 'py, T>` has two lifetime parameters `'a` and `'py`.
```

Comment:
If we make a point here about these, we should also explain _what_ these lifetimes describe, not just that they are there.
What does `'py` describe? Is it the same for `Bound` and `Borrowed`? What about `'a`? 

Or we remove this here and defer to the detail sections below.

Reply:
Tried to expand on this a bit.